### ISSUE TITLE
[SideSheet] - 1.10.0-alpha01 - NPE: shapeAppearanceModel not initialized before layout (with workaround)

### ISSUE THREAD

**Issue by fat-tire** [material-components/material-components-android#3322]
Runtime NPE when trying to layout a SideSheet (1.10-alpha01):

```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.material.shape.ShapeAppearanceModel$Builder com.google.android.material.shape.ShapeAppearanceModel.toBuilder()' on a null object reference
      	at com.google.android.material.sidesheet.SideSheetBehavior.setSheetEdge(SideSheetBehavior.java:163)
     	at com.google.android.material.sidesheet.SideSheetBehavior.setSheetEdge(SideSheetBehavior.java:158)
  	at com.google.android.material.sidesheet.SideSheetBehavior.onLayoutChild(SideSheetBehavior.java:352)
     	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
```

Line 163 is where it crashes:
```kotlin
      ShapeAppearanceModel.Builder builder = shapeAppearanceModel.toBuilder();
```

Turns out `shapeApparanceModel` is null.  Hence the crash.

`shapeApparanceModel` should have been set on line 134 earlier:
```kotlin
    if (a.hasValue(R.styleable.SideSheetBehavior_Layout_shapeAppearance)) {
      this.shapeAppearanceModel =
          ShapeAppearanceModel.builder(context, attrs, 0, DEF_STYLE_RES).build();
    }
```

But the assignment is skipped as the test w/ `a` fails.  Let's look at where `a` is set:

Line 128:
```kotlin
    TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.SideSheetBehavior_Layout);
```

We need to make sure `a.hasValue(R.styleable.SideSheetBehavior_Layout_shapeAppearance`).

Here's my workaround to set `a` so that it won't fail the test on 134.  In `app/src/main/res/values/themes.xml` (or `style.xml`), add something like this to the app's main `<style>`:

```xml
        <item name="shapeAppearance">@style/Widget.Material3.SideSheet</item>
```

`Widget.Material3.SideSheet.Modal` also works.  For example:
```xml
<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:tools="http://schemas.android.com/tools">
    <!-- the theme applied to the application or activity -->
    <style name="Theme.MyApp" parent="Theme.Material3.Dark">
         ...
         <item name="shapeAppearance">@style/Widget.Material3.SideSheet</item>
    </style>
</resources>
```

Not sure if this will work for every configuration, in every case... but it worked in my quick little test, including the new left edge stuff.  Great job, @afohrman!

**Comment by fat-tire** [comment]
I'm gonna add a second potential bug here- this one without a solution.

If you have a sidesheet with some view (`R.id.myOtherView`) as the coplanarSiblingViewID:

`SideSheetBehavior.from(mysidesheet).setCoplanarSiblingViewId(R.id.myOtherView)`

You open the sidesheet on a landscape orientation phone, or maybe a tablet or Chromebook or some other big screen:

`SideSheetBehavior.from(mysidesheet).expand()`

Great, it shoves `myOtherView` over (squashes it) for a side by side!  But should you then switch to portrait, or resize the window/activity to be very narrow, you'd want want to go from a side-by-side to a regular "on top" sidesheet.

So you react to the change with either:

`SideSheetBehavior.from(mysidesheet).setCoplanarSiblingViewId(View.NO_ID)`

or alternately:

`SideSheetBehavior.from(mysidesheet).setCoplanarSiblingView(null)`

_What should happen:_

The sidesheet stays expanded, changes type to non-coplanar regular, and the formerly coplanar view (`myOtherView`) should expand "underneath it".

_What does happen:_

The sidesheet stays expanded, changes type to non-coplanar regular behavior, but leaves the other view (`myOtherView`) squashed.  Hiding the stylesheet at this point makes it go away, but the coplanar (`myOtherView`) view doesn't expand and it doesn't appear behind it.  It just sits there, squished, with a big grey rectangle sitting there where the sidesheet was.

_What doesn't work to fix it:_

Doing any combination of `invalidate()`, `requestLayout()` or `forceLayout()` on `myOtherView` to get it to expand back into the grey space.

**tl;dr** -- when changing from coplanar to standard sidesheet (by setting `setCoplanarSiblingViewId()`), if the sidesheet is already open, the squashed side view does not expand underneath it as expected, leaving a big gray rectangle under the regular sidesheet.

Thx!

**Comment by pekingme** [comment]
Thanks for actively working on testing out the side sheet! 
Do you mind to file a different issue with your second comment?

**Comment by fat-tire** [comment]
Sure I can do that.  I figured since I had this assigned I'd just combine them, but I'll do a new one right now.
